Documentation/ApiOverview/FeatureToggles/Index.rst

```diff
@@ -0,0 +1,81 @@
+.. include:: ../../Includes.txt
+
+
+
+.. _feature-toggles:
+
+Feature Toggles
+===============
+
+TYPO3 provides an API class for creating so-called `Feature Toggles`. Feature toggles provide an easy way to add
+new implementations of features next to their legacy version. By using a feature toggle, the integrator or site admin
+can decide when to switch to the new feature.
+
+The API checks against a system-wide option array within :php:`$TYPO3_CONF_VARS['SYS']['features']` which an integrator
+or admin can set in the :file:`LocalConfiguration.php` file.
+Both TYPO3 Core and Extensions can provide alternative functionality for a certain feature.
+
+Examples for features are:
+
+ - Throw exceptions in new code instead of just returning a string message as error message.
+ - Disable obsolete functionality which might still be used, but slows down the system.
+ - Enable alternative PageNotFound handling for an installation.
+
+Naming of Feature Toggles
+-------------------------
+
+Feature names should NEVER be named "enable" or have a negation, or contain versions or years.
+
+Bad examples: 
+
+ - `enableFeatureXyz`
+ - `disableOverlays`
+ - `schedulerRevamped2018`
+ - `useDoctrineQueries`
+ - `disablePreparedStatements`
+ - `disableHooksInFE`
+
+Good examples:
+
+ - `ExtendedRichtextFormat`
+ - `NativeYamlParser`
+ - `InlinePageTranslations`
+ - `TypoScriptParserIncludesAsXml`
+ - `NativeDoctrineQueries`
+
```

Comment:
In general we would have to write something about namespaces as well, since Extensions also might make use of feature toggles - e.g. `<extension>.<feature-path>` or `<package>.<feature-path>`

Reply:
See ll. 58 - for extensions there is the hint to prefix with the extension key. but we could add it here additionally.